subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1707,19 +1719,34 @@ static int ztls_poll_update_ctx(struct net_context *ctx,
 	}
 
 	if (pfd->events & ZSOCK_POLLIN) {
```

Comment:
Then we `assume that socket is always writable`. POLLOUT is not an issue in this case.

Reply:
> Then we `assume that socket is always writable`.

But a socket is not always writable. For normal socket, this is well known issue (well, bug) present so far because resolving it would require adding more overhead to socket structures, and which is properly ticketed as https://github.com/zephyrproject-rtos/zephyr/issues/5842.

Based on the response above, I don't see that this issue was even considered/analyzed for TLS sockets, far less in queue for fixing.